extension/src/router_channel.rs

```diff
@@ -251,7 +245,7 @@ impl RouterChannel {
       let (mut app_req_tx, app_req_recv) = mpsc::channel::<Result<Frame<Bytes>>>(32 * 1024);
       let app_req = app_req_builder.body(StreamBody::new(app_req_recv))?;
 
-      while futures::future::poll_fn(|ctx| app_sender.poll_ready(ctx))
+      if futures::future::poll_fn(|ctx| app_sender.poll_ready(ctx))
```

Comment:
Same here... I think I copied this out of the hyper docs or examples or out of the `oha` app... I've not read the `poll_fn` docs in detail to confirm if this is going to change the behavior here.